http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/MultiplexedChannelRecord.java

```diff
@@ -184,7 +185,19 @@ void closeChildChannels() {
      * Delivers the exception to all registered child channels, and prohibits new streams being created on this connection.
      */
     void closeChildChannels(Throwable t) {
-        closeAndExecuteOnChildChannels(ch -> ch.pipeline().fireExceptionCaught(t));
+        closeAndExecuteOnChildChannels(ch -> {
+            RuntimeException ioException = new RuntimeException("An exception occurred on the connection ", t);
```

Comment:
`RuntimeException` might not be the best exception to be wrapped with. The reason I didn't use `IOException` is because I figured not all exceptions thrown on connection are retryable, but is that true?

Reply:
Wouldn't we want IOExceptions to stay IOExceptions? Any reason we need to re-wrap here?

Reply:
The reason I re-wrap it here is to provide more descriptive message on the exception. Currently, it's not very clear why the request fails.

Reply:
Wouldn't this make currently-retried exceptions not be retried? Should we make it an IOException if the wrapped is an IOException?

Reply:
I thought our default retry policy would check if the root cause of the exception is assignable from `IOException`, but turns out it would check one level down. 

https://github.com/aws/aws-sdk-java-v2/blob/138def267fc8814418e5b7c8ab80f038b9d40641/core/sdk-core/src/main/java/software/amazon/awssdk/core/retry/conditions/RetryOnExceptionsCondition.java#L59-L62

I'll make it an IOException if the wrapped is an IOException

Reply:
Or we could update the retry policy to check down the entire chain? That would probably be better.

Reply:
+1. I'll create a backlog item for it. I think it'd still be nice to wrap IOException with an IOException here to make it clear this is a transient retryable I/O related error.